(Naga) Add support for Descriptor Runtime Indexing when ingesting SPIR-V

hasenbanck opened the pull request:
**Connections**

Fixes #6423

**Description**
NAGA is already able to ingest SPIR-V that uses descriptor runtime indexing and handles them as binding arrays correctly, when outputing WGSL. I only added the missing capabilities so that NAGA can now properly ingest SPIR-V with descriptor runtime indexing. My main use case is to be able to use the Slang shader language via SPIR-V.

**Testing**
I added another snapshot test that uses the "OpCapability RuntimeDescriptorArray" and is successfully compiled into an WGSL shader that implements binding arrays properly.

I also successfully tested this change in our game ingesting SLANG shaders, which use runtime binding arrays, compiled to SPIR-V into WGPU.

**Squash or Rebase?**
Rebase

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy --tests`.
- [x] Run `cargo xtask test` to run tests.
- [x] If this contains user-facing changes, add a `CHANGELOG.md` entry.

cwfitzgerald commented on naga/tests/in/spv/binding-arrays.runtime.spv:
We shouldn't be checking in the raw spirv, just the asm.

cwfitzgerald commented on naga/tests/in/spv/binding-arrays.runtime.spvasm:
Could you also drop the slang file in here? So we can reproduce this if we want to.

cwfitzgerald reviewed (approved):
Some minor comments, then we're good to land.

hasenbanck commented:
I removed the SPV and added the Slang source instead. The disassembled ASM changed a bit, but the final WGSL is the same.

Edit: Also rebased the branch to be mergable again.

hasenbanck committed:
(Naga) Add support for Descriptor Runtime Indexing when ingesting SPIR-V

cwfitzgerald reviewed (approved):
Nice!
